README.md

```diff
@@ -14,3 +13,103 @@ provided by the bot. You will only need to do this once across all repos using o
+
+![License](https://img.shields.io/badge/license-MIT-green.svg)
+![Docker Image Build](https://github.com/retaildevcrews/helium-typescript/workflows/Docker%20Image%20Build/badge.svg)
+
+This is a Java Spring Boot Web API reference application designed to "fork and code" with the following features:
+
+- Securely build, deploy and run an App Service (Web App for Containers) application
+- Use Managed Identity to securely access resources
+- Securely store secrets in Key Vault
+- Securely build and deploy the Docker container from Container Registry
+- Connect to and query CosmosDB
+- Automatically send telemetry and logs to Azure Monitor
+- Instructions for setting up Key Vault, ACR, Azure Monitor and Cosmos DB are in the Helium [readme](https://github.com/retaildevcrews/helium)
+
+## Prerequisites
+
+- Azure subscription with permissions to create:
+  - Resource Groups, Service Principals, Keyvault, CosmosDB, App Service, Azure Container Registry, Azure Monitor
+- Bash shell (tested on Mac, Ubuntu, Windows with WSL2)
+  - Will not work in Cloud Shell unless you have a remote dockerd
+- Azure CLI 2.0.72+ ([download](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest))
+- Docker CLI ([download](https://docs.docker.com/install/))
+- Java 8 above ([download](https://nodejs.org/en/download/)) 
+- JQ ([download](https://stedolan.github.io/jq/download/))
+- Completion of the Key Vault, ACR, Azure Monitor and Cosmos DB setup in the Helium [readme](https://github.com/retaildevcrews/helium)
+- Visual Studio Code (optional) ([download](https://code.visualstudio.com/download))
+
+### Dependency Vulnerability
+
+Any dependencies?  
+
+## Setup
+
+The application requires Key Vault and Cosmos DB to be setup per the Helium [readme](https://github.com/retaildevcrews/helium)
+
+### Run locally on your machine
+
+- Fork this repo and clone to your local machine  https://github.com/retaildevcrews/helium-java.git 
+  - All instructions assume starting from the root of the repo
+
+```bash
+# run locally with unit tests
+mvn clean package
+
+# run locally without unit tests
+mvn clean package -DskipTests
+
+mvn clean spring-boot:run
+Could we use mvn clean spring-boot:start here to run in the background?
+Then we could shut it down more cleanly with a mvn spring-boot:stop
+
+# test the application (takes about 10 seconds to start)
+curl http://localhost:8080/healthz
+
+# Stop the application by typing Ctrl-C in the terminal window
+Or :::::
+mvn spring-boot:stop
+
+### Run application as a local container instead
+
+```bash
+
+# Make sure you are in the root of the repo
+# Build the container first 
+## Run this docker-dev command to build an alpine image with Azure CLI installed in the container OR run the docker build command
+docker build . -t docker-dev
+docker build -t myimage_name .
+
+# Then run the container
+docker run  -p8080:8080 --env KEYVAULT_NAME=devshop-gelatodev-kv myimage_name:latest
+
```

Comment:
this should be port 4120 - or update the instructions below

Reply:
The default for Spring is 8080.  I've added a task to change it for Helium Java: https://github.com/retaildevcrews/helium-java/issues/181 .

For this, we should use 8080 and update the documentation to 4120 as part of the task I linked above.

Reply:
8080 is fine as long as you update the docs. 4120 is fine too

Reply:
Updated to 4120